src/wasm-lib/kcl/src/parser/parser_impl.rs

```diff
@@ -0,0 +1,2045 @@
+            let source_ranges = vec![SourceRange([part.start(), part.end()])];
+            let val = match part {
+                // TODO: these should be valid operands eventually,
+                // users should be able to run "let x = f() + g()"
+                // see https://github.com/KittyCAD/modeling-app/issues/783
+                Value::FunctionExpression(_)
+                | Value::PipeExpression(_)
+                | Value::PipeSubstitution(_)
+                | Value::ArrayExpression(_)
+                | Value::ObjectExpression(_) => {
+                    return Err(KclError::Syntax(KclErrorDetails {
+                        source_ranges,
+                        message: TODO_783.to_owned(),
+                    }))
+                }
+                Value::UnaryExpression(x) => BinaryPart::UnaryExpression(x),
+                Value::Literal(x) => BinaryPart::Literal(x),
+                Value::Identifier(x) => BinaryPart::Identifier(x),
+                Value::BinaryExpression(x) => BinaryPart::BinaryExpression(x),
+                Value::CallExpression(x) => BinaryPart::CallExpression(x),
+                Value::MemberExpression(x) => BinaryPart::MemberExpression(x),
+            };
+            Ok(val)
+        })
+        .context(expected("an operand (a value which can be used with an operator)"))
+        .parse_next(i)?;
+    Ok(op)
+}
+
+impl TokenType {
+    fn parse_from(self, i: TokenSlice) -> PResult<Token> {
+        any.try_map(|token: Token| {
+            if token.token_type == self {
+                Ok(token)
+            } else {
+                Err(KclError::Syntax(KclErrorDetails {
+                    source_ranges: token.as_source_ranges(),
+                    message: format!(
+                        "expected {self} but found {} which is a {}",
+                        token.value.as_str(),
+                        token.token_type
+                    ),
+                }))
+            }
+        })
+        .parse_next(i)
+    }
+}
+
+/// Parse some whitespace (i.e. at least one whitespace token)
+fn whitespace(i: TokenSlice) -> PResult<Vec<Token>> {
+    repeat(
+        1..,
+        any.try_map(|token: Token| {
+            if token.token_type == TokenType::Whitespace {
+                Ok(token)
+            } else {
+                Err(KclError::Syntax(KclErrorDetails {
+                    source_ranges: token.as_source_ranges(),
+                    message: format!(
+                        "expected whitespace, found '{}' which is {}",
+                        token.value.as_str(),
+                        token.token_type
+                    ),
+                }))
+            }
+        }),
+    )
+    .context(expected("some whitespace (e.g. spaces, tabs, new lines)"))
+    .parse_next(i)
+}
+
+/// Parse the = operator.
+fn equals(i: TokenSlice) -> PResult<Token> {
+    any.verify(|token: &Token| matches!(token.token_type, TokenType::Operator) && token.value == "=")
```

Comment:
Couldn't your `any.verify`s be `one_of`?

Reply:
Again, winnow-rs/winnow#346 -- my token does not impl `AsChar` which is required by most of `ContainsToken`.